### PR TITLE
Ftrack ID from  Flashtalking by Mediaocean RTD submodule: initial release

### PR DESCRIPTION
**Pull request by ftxmoJason**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Adding a new Real Time Data submodule for ftrack

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Comment by patmmccann** [comment]
Appears this should be an id submodule?

File: modules/ftrackRtdProvider.js [hunk 1]
```diff
@@ -0,0 +1,691 @@
...
+
+    FtBh.prototype = {
+      get: function get() {
+        var ua = navigator.userAgent.toLowerCase();
+        var keys = [];
+        var navLang = navigator.language || navigator.browserLanguage;
+        var navLangArr = navLang.split('-');
+
+        if (typeof navLangArr[0] == 'undefined') {
+          // navLang = navLang;
```
Comment (Rothalack) [inline modules/ftrackRtdProvider.js:164]: Is this intended? You could do

```
if (typeof navLangArr[0] !== 'undefined') {
  navLang = navLangArr[0];
}
```

**Comment by Rothalack** [comment]
The other thing I have noticed in testing is according to ftrackRtdProvider.md you expect the POST payload of https://grid.bidswitch.net/hbjson to be the user object with data, name, segment, etc. When I set this up in testing, I only got a user object with an ID. Not sure if that's a mismatch of documentation to intended behavior or if I did something incorrect in testing. This is the object I get. Let me know if that's correct. Otherwise, I think this is good to go.
{
    "id": "18619f5c463f47",
    "site": {
        "page": "http%3A%2F%2Flocalhost%3A9999%2FintegrationExamples%2Fgpt%2Fhello_world.html%3Fpbjs_debug%3Dtrue"
    },
    "tmax": 1000,
    "source": {
        "tid": "7cd45a25-91d1-46f0-b669-7bae5ad024b3",
        "ext": {
            "wrapper": "Prebid_js",
            "wrapper_version": "6.4.0-pre"
        }
    },
    "imp": [
        {
            "id": "2a8cf2c7acae99",
            "tagid": "12341234",
            "ext": {
                "divid": "div-gpt-ad-1460505748561-0"
            },
            "banner": {
                "w": 300,
                "h": 250,
                "format": [
                    {
                        "w": 300,
                        "h": 250
                    },
                    {
                        "w": 300,
                        "h": 600
                    }
                ]
            }
        }
    ],
    "user": {
        "id": "5901ec58c7804216bc5ee8c2b0ea92db"
    }
}

**Comment by ftxmoJason** [comment]
@Rothalack I can't reply to your comment here for some reason, but the `user` property would not have the ftrack ID until https://github.com/prebid/Prebid.js/pull/7813 is merged in. For now, if you lookin in the localStorage, you will see the ID stored there. Please see: DevTools > Application > Local Storage > Your WebPage: You will see `ftrack-rtd` and `ftrack-rtd_exp` in the storage. If those exists than this module is working as expected.

File: modules/ftrackRtdProvider.js (hunk 1 above)
Comment (ftxmoJason) [reply modules/ftrackRtdProvider.js:164]: Thanks @Rothalack , I can make this change.

Comment (ftxmoJason) [reply modules/ftrackRtdProvider.js]: Ok, this has been committed.

**Comment by patmmccann** [comment]
@ftxmoJason I am planning to close this PR unless you respond to my question above. Thank you.

**Comment by ftxmoJason** [comment]
@patmmccann Thanks for the question Pat. Sorry for the slow response to your q, I wanted to make sure the folks who assigned this project to me answered your question for me:

The FTrack solution is not limited to associating a single identifier per user, thus we consider it data associated with a user and not necessarily a unique identifier. Our plan is to extend identifiers to several segments; hence the RTD module is better suited for our scenarios.

**Comment by patmmccann** [comment]
we'll have to discuss in committee, this is a bit of an edge case

@gglas fyi

**Comment by ftxmoJason** [comment]
Ok, thanks @patmmccann . If you need any further clarification, let us know!

**Comment by ftxmoJason** [comment]
Hello @patmmccann 
We have update the code to add an extension to the Ftrack rtd module in order to include the more segment data.

**Comment by patmmccann** [comment]
Thanks! @ftxmoJason please note the lgtm error

File: modules/ftrackRtdProvider.js [hunk 2]
```diff
@@ -0,0 +1,713 @@
...
+        if (!obj) {
+          try {
+            obj = new window.ActiveXObject('PDF.PdfCtrl');
+          } catch (e) {
+            obj = null;
+            return null;
+          }
+        }
+
+        if (obj !== null) {
```
Comment (patmmccann) [inline modules/ftrackRtdProvider.js:414]: this is always true, throwing error on lgtm.com

Comment (ftxmoJason) [reply modules/ftrackRtdProvider.js]: Thanks @patmmccann .  PR update incoming.

**Comment by patmmccann** [comment]
@ftxmoJason I still; I think it is a mistake for an id provider to not set their id on the eids object, as it will severely limit ssp uptake. It currently seems like flashtalking is only interested in iponweb reading this value. It also appears to be at least one rules violation https://docs.prebid.org/dev-docs/module-rules.html: "RTD modules must not supply privacy-sensitive user information (including IDs) into the auction. That functionality is reserved for User ID modules." and perhaps also rules around fairness to bidders as only one bidder expects an id in this location. I think this module can move forward with the segments but that you should pass the id on a second module and encourage your publisher partners to install both.

**Pull request closed without merging**